discord/voice_client.py

```diff
@@ -56,9 +58,89 @@
 from . import opus
 from .backoff import ExponentialBackoff
 from .gateway import *
-from .errors import ClientException, ConnectionClosed
+from .errors import ClientException, ConnectionClosed,  InvalidVoicePacket
 from .player import AudioPlayer, AudioSource
 
+class VoicePacket:
+    """Represents a packet of voice data.
+
+    You should not construct these yourself, they will be constructed from
+    incoming UDP packets on the socket used for voice data.
+
+    Attributes
+    ----------
+    sequence : int
+        The sequence id for the packet. Must be between 0 and 65535. Each
+        packet has a value of 1 greater than the previous packet from the
+        same source. After the sequence reaches the maximum value (65535),
+        the next packet has a sequence value of 0.
+    timestamp : int
+        The timestamp for the packet.
+    ssrc : int
+        The id of the user (or source) of this voice packet.
+    buff : bytes
+        The Opus packet (as bytes).
+    user : Optional[:class:`abc.User`]
+        The user who the packet belongs to
+    pcm : Optional[bytes]
+        The decoded PCM data.
+    """
+    # Struct format of the packet metadata
+    _FORMAT = '>HHII'
+    # Packets should start with b'\x80\x78' (32888 as a big endian ushort)
+    _CHECK = 32888
+    # Some packets will start with b'\x90\x78' (36984)
+    _CHECK2 = 36984
+
+    def __init__(self, sequence=None, timestamp=None, ssrc=None, buff=None, user=None):
```

Comment:
a lot of these will probably be constructed. maybe it'd be advantageous to make this a slotted class.